samcli/lib/sync/watch_manager.py

```diff
@@ -120,10 +120,24 @@ def _add_code_triggers(self) -> None:
                 continue
             self._observer.schedule_handlers(trigger.get_path_handlers())
 
-    def _add_template_trigger(self) -> None:
+    def _add_template_triggers(self) -> None:
         """Create TemplateTrigger and add its handlers to observer"""
-        template_trigger = TemplateTrigger(self._template, lambda _=None: self.queue_infra_sync())
-        self._observer.schedule_handlers(template_trigger.get_path_handlers())
+        stacks = SamLocalStackProvider.get_stacks(self._template)[0]
+        for stack in stacks:
+            template = stack.location
+            try:
+                template_trigger = TemplateTrigger(template, stack.name, lambda _=None: self.queue_infra_sync())
+            except InvalidTemplateFile:
+                LOG.debug(
+                    "Template %s not watched due to template file validation failed for stack %s.",
+                    template,
+                    stack.name
+                )
+                continue
+
+            if not template_trigger:
+                continue
+            self._observer.schedule_handlers(template_trigger.get_path_handlers())
```

Comment:
I would move these 3 lines inside try/except block so that it will be more easy to follow-up. And what is the reason that `template_trigger` might be `None`?

Reply:
The structure is following the add code triggers method above. Yes there is no reason yet that a trigger can be none. It can be removed.